src/PreCICE.jl

```diff
@@ -1332,5 +1340,149 @@ function actionReadIterationCheckpoint()
     return unsafe_string(msgCstring)
 end
 
+@doc """
+
+        isGradientDataRequired(dataID::Integer)
+        
+Checks if the given data set requires gradient data. We check if the data object has been intialized with the gradient flag.
+# Arguments
+- `dataID::Integer`: ID of the data to be checked. Obtained by [`getDataID`](@ref).
+
+"""
+function isGradientDataRequired(dataID::Integer)
+    return ccall((:precicec_isGradientDataRequired, "libprecice"), Cbool, dataID)
+end
+
+@doc """
+
+writeBlockVectorGradientData(dataID::Integer, valueIndices::AbstractArray{Cint}, gradientValues::AbstractArray{Float64})
+
+
+Write gradient data of a vector data as a block, the value of a specified vertices to a dataID.
+
+# Arguments
+- `dataID::Integer`: ID of the data to be written. Obtained by [`getDataID`](@ref).
+- `valueIndices::AbstractArray{Cint}`: Indices of the vertex.
+- `gradientValues::AbstractArray{Float64}`: The gradient values to write.
+
+# Notes
+
+Previous calls:
+ - [`initialize`](@ref) 
+
+# Examples
+
+Write gradient values of a vector data for a 2D problem with 2 vertices:
+
+```julia
+data_id = 1
+valueIndices = [1,2,3]
+gradientValues = [1.0, 2.0, 3.0, 4.0, 5.0, 6.0, 7.0, 8.0]
+writeBlockVectorGradientData(data_id, vertex_id, gradientValue)
```

Comment:
`replace vertex_id` with `valueIndices` or the other way around. `s` missing in gradientValue. Are the gradient values 1D or 2D? The number of Values doesn't match in either case to the indices, but if it's 2D, we have to make a disclaimer, that the values have to be flattened. We usually try to have 2D values also als a Nx2 Matrix.

Reply:
If it is correct, you should give the Format in which the Values should be given